### PR TITLE
Added Inventory

### PR DESCRIPTION
**Pull request by MercyfulFate**
Inventory System.

I've added documentation to the "client/inventory" code that expains things. everything still has the mf_\* prefix (my obfuscator uses it).

I used my mf_comile function for most of the client_compiles... when running #define DEBUG true in init.sqf (the main mission init.sqf" it will dynamically recompile the sqf files each call, so much much less restarting the mission when changing something small. if you set DEBUG false then it will compileFinal the functions instead, so for release  please dont forget :P

### CODE REVIEW

**Comment by MercyfulFate** [comment]
By the way if you want to change any item's ground object/deploy object types that are used (some of them are just used because there is no good alternative atm) I'll list the locations....

Just to note:
ground type => the object you get when you "drop" the item. It can be picked up instantly and has no other effects.
deploy type => the object that is created when an item (e.g spawn-beacon) is deployed. item effects (eg, spawn button in the repawn menu) are only active while deployed, and the object must be packed up to be moved.

Spawn Beacon:
deployed type: Line 13 in "client/items/beacon/init.sqf". currently a campfire atm
ground type: Line 17 in "client/items/beacons/init.sqf". currently a briefcase

Medkit:
ground type: Line 12 in "client/items/misc/init.sqf". currently a briefcase

Repair Kit:
ground type: line 16 in "client/items/misc/init.sqf". currently a briefcase

Improv Roof:
ground type: line 12 in "client/items/camonet/init.sqf". currently a briefcase

The others I assume you wont want to change.

On a side note.... enemies are able to steal spawn beacons :)

File: client/items/jerrycan/can_refuel.sqf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+#define ERR_NO_FUEL "You have no full fuel cans"
+#define ITEM_COUNT(ITEMID) ITEMID call mf_inventory_count
+private ["_vehicle", "_error"];
+_vehicle = objNull;
+if (count _this == 0) then { // if array empty
+    _vehicle = call mf_jerrycan_nearest_vehicle;
+} else {
+    _vehicle = _this select 0;
+};
+
```
Comment (JoSchaap) [inline client/items/jerrycan/can_refuel.sqf:21]: shouldnt we check if vehicle is unlocked, canmove and damage below 1 first? :)  same for repairs and else state that repair refuel is not available for this vehicle

File: client/systems/generalStore/buyItems.sqf [hunk 2]
```diff
@@ -108,15 +96,15 @@ for [{_x=0},{_x<=_size},{_x=_x+1}] do
...
         };
         case "Improv. roof": {
-            if(((player getVariable "camonet") + 1 <= 1) AND ((player getVariable "camonet") + 1 <= 1)) then {
-                player setVariable["camonet",(player getVariable "camonet") + 1,true];
+			if not(MF_ITEMS_CAMO_NET call mf_inventory_is_full) then {
+				[MF_ITEMS_CAMO_NET, 1] call mf_inventory_add;
             } else {
             	_price = 0;
-                {if(_x select 0 == "Improv. roof") then{_price = _x select 4;};}forEach (call generalStore);
+                {if(_x select 0 == "Improv. roof") then{_price = _x select 4;};}forEach generalStore;
```
Comment (JoSchaap) [inline client/systems/generalStore/buyItems.sqf:103]: this call should stay since the stores contents are pre-compiled

File: init.sqf [hunk 3]
```diff
@@ -5,6 +5,7 @@
 //	@file Description: The main init.
 
 #include "setup.sqf"
+#define DEBUG true
```
Comment (JoSchaap) [inline init.sqf:8]: note to self: turn this off :)

File: client/systems/generalStore/buyItems.sqf [hunk 4]
```diff
@@ -69,38 +68,27 @@ for [{_x=0},{_x<=_size},{_x=_x+1}] do
...
-            	if (!((player getVariable "fuelFull") + 1 <= 1)) then {
-	            	_price = 0;
-	                {if(_x select 0 == "Jerry Can (Full)") then{_price = _x select 4;};}forEach (call generalStore);
-	            	genStoreCart = genStoreCart - _price;    
-                } else {
-                    player setVariable["fuelEmpty",0,true];
-                    player setVariable["fuelFull",1,true];
-                };
+				_price = 0;
+				{if(_x select 0 == "Jerry Can (Full)") then{_price = _x select 4;};}forEach generalStore;
```
Comment (JoSchaap) [inline client/systems/generalStore/buyItems.sqf:75]: same here: re-add (call  :)

File: client/systems/generalStore/buyItems.sqf [hunk 5]
```diff
@@ -69,38 +68,27 @@ for [{_x=0},{_x<=_size},{_x=_x+1}] do
...
-            	if (((player getVariable "fuelFull") + 1 <= 1)) then {
-	            	_price = 0;
-	                {if(_x select 0 == "Jerry Can (Empty)") then{_price = _x select 4;};}forEach (call generalStore);
-	            	genStoreCart = genStoreCart - _price;    
-                } else {
-                    player setVariable["fuelEmpty",1,true];
-                    player setVariable["fuelFull",0,true];
-                };
+            	_price = 0;
+                {if(_x select 0 == "Jerry Can (Empty)") then{_price = _x select 4;};}forEach generalStore;
```
Comment (JoSchaap) [inline client/systems/generalStore/buyItems.sqf:85]: re-add (call

**Pull request merged**